src/schneider_lab_to_nwb/zempolich_2024/zempolich_2024_convert_all_sessions.py

```diff
@@ -0,0 +1,164 @@
+    """
+    data_dir_path = Path(data_dir_path)
+    output_dir_path = Path(output_dir_path)
+    session_to_nwb_kwargs_per_session = get_session_to_nwb_kwargs_per_session(data_dir_path=data_dir_path)
+
+    futures = []
+    with ProcessPoolExecutor(max_workers=max_workers) as executor:
+        for session_to_nwb_kwargs in session_to_nwb_kwargs_per_session:
+            session_to_nwb_kwargs["output_dir_path"] = output_dir_path
+            session_to_nwb_kwargs["verbose"] = verbose
+            nwbfile_name = get_nwbfile_name_from_kwargs(session_to_nwb_kwargs)
+            exception_file_path = output_dir_path / f"ERROR_{nwbfile_name}.txt"
+            futures.append(
+                executor.submit(
+                    safe_session_to_nwb,
+                    session_to_nwb_kwargs=session_to_nwb_kwargs,
+                    exception_file_path=exception_file_path,
+                )
+            )
+        for _ in tqdm(as_completed(futures), total=len(futures)):
+            pass
+
+
+def get_nwbfile_name_from_kwargs(session_to_nwb_kwargs):
+    behavior_file_path = session_to_nwb_kwargs["behavior_file_path"]
+    subject_id = behavior_file_path.name.split("_")[1]
+    session_id = behavior_file_path.name.split("_")[2]
+    nwbfile_name = f"sub-{subject_id}_ses-{session_id}.nwb"
+    return nwbfile_name
+
+
+def safe_session_to_nwb(*, session_to_nwb_kwargs: dict, exception_file_path: str | Path):
+    """Convert a session to NWB while handling any errors by recording error messages to the exception_file_path.
+
+    Parameters
+    ----------
+    session_to_nwb_kwargs : dict
+        The arguments for session_to_nwb.
+    exception_file_path : Path
+        The path to the file where the exception messages will be saved.
+    """
+    exception_file_path = Path(exception_file_path)
+    try:
+        session_to_nwb(**session_to_nwb_kwargs)
+    except Exception as e:
+        with open(exception_file_path, mode="w") as f:
+            f.write(f"session_to_nwb_kwargs: \n {pformat(session_to_nwb_kwargs)}\n\n")
+            f.write(traceback.format_exc())
+
+
+def get_session_to_nwb_kwargs_per_session(
+    *,
+    data_dir_path: str | Path,
+):
+    """Get the kwargs for session_to_nwb for each session in the dataset.
+
+    Parameters
+    ----------
+    data_dir_path : str | Path
+        The path to the directory containing the raw data.
+
+    Returns
+    -------
+    list[dict[str, Any]]
+        A list of dictionaries containing the kwargs for session_to_nwb for each session.
+    """
+    a1_ephys_path = data_dir_path / "A1_EphysFiles"
+    a1_ephys_behavior_path = data_dir_path / "A1_EphysBehavioralFiles"
+    a1_opto_path = data_dir_path / "A1_OptoBehavioralFiles"
+    m2_ephys_path = data_dir_path / "M2_EphysFiles"
+    m2_ephys_behavior_path = data_dir_path / "M2_EphysBehavioralFiles"
+    m2_opto_path = data_dir_path / "M2_OptoBehavioralFiles"
+
+    a1_kwargs = get_brain_region_kwargs(
```

Comment:
OK, so you have sessions that correspond to A1 area and sessions that correspond to M2 area. You gather each session kwargs separately and append them. Make sense to me.